From bf4150b885726c0f621b9ad013db5d64cb8d8db0 Mon Sep 17 00:00:00 2001
From: Kishore Kadiyala <kishore.kadiyala@intel.com>
Date: Mon, 7 Mar 2016 10:43:40 +0530
Subject: [PATCH 2/2] input: touch: silead: Fix for possible case of Null
 Pointer dereference

There is a possible case of Null pointer dereference in probe with
Pointer 'data' returned from call to function 'devm_kzalloc'.

Change-Id: Ie9026e99ed8c117d65d64c1d7ea6b8faf93ae38c
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-22956
Signed-off-by: Kishore Kadiyala <kishore.kadiyala@intel.com>
Reviewed-on: https://android.intel.com:443/478119
---
 drivers/input/touchscreen/silead.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/input/touchscreen/silead.c b/drivers/input/touchscreen/silead.c
index 9c96893..d3acdd7 100644
--- a/drivers/input/touchscreen/silead.c
+++ b/drivers/input/touchscreen/silead.c
@@ -622,6 +622,8 @@ static int silead_ts_probe(struct i2c_client *client,
 	}
 
 	data = devm_kzalloc(dev, sizeof(*data), GFP_KERNEL);
+	if (!data)
+		return -ENOMEM;
 	i2c_set_clientdata(client, data);
 	data->client = client;
 
-- 
1.9.1

